core/include/usBundle.h

```diff
@@ -47,143 +48,297 @@ typedef ServiceReference<void> ServiceReferenceU;
    *
-   * @see \ref MicroServices_BundleProperties
+   * @param rhs The \c Bundle object to compare this object with.
+   * @return Returns the result of <code>!(*this == rhs)</code>.
    */
-  static const std::string PROP_LOCATION;
+  bool operator!=(const Bundle& rhs) const;
 
   /**
-   * The property key for looking up this bundle's version identifier.
-   * The property value is of type \c std::string.
+   * Compares this \c Bundle with the specified bundle for order.
+   *
+   * %Bundle objects are ordered first by their framework id and
+   * then according to their bundle id.
+   * Invalid \c Bundle objects will always compare greater then
+   * valid \c Bundle objects.
    *
-   * @see \ref MicroServices_BundleProperties
+   * @param rhs The \c Bundle object to compare this object with.
+   * @return \c
    */
-  static const std::string PROP_VERSION;
+  bool operator<(const Bundle& rhs) const;
 
   /**
-   * The property key for looking up this bundle's vendor information.
-   * The property value is of type \c std::string.
+   * Tests this %Bundle object for validity.
    *
-   * @see \ref MicroServices_BundleProperties
+   * Invalid \c Bundle objects are created by the default constructor or
+   * can be returned by certain framework methods if the bundle has been
+   * uninstalled.
+   *
+   * @return \c true if this %Bundle object is valid and can safely be used,
+   *         \c false otherwise.
    */
-  static const std::string PROP_VENDOR;
+  bool IsValid() const;
 
   /**
-   * The property key for looking up this bundle's description.
-   * The property value is of type \c std::string.
-   *
-   * @see \ref MicroServices_BundleProperties
+   * Boolean conversion operator using IsValid().
    */
-  static const std::string PROP_DESCRIPTION;
+  explicit operator bool() const;
 
-  virtual ~Bundle();
+  /**
+   * Releases any resources held or locked by this
+   * \c Bundle and renders it invalid.
+   */
+  Bundle& operator=(std::nullptr_t);
 
   /**
    * Returns this bundle's current state.
    *
-   * <p>
    * A bundle can be in only one state at any time.
    *
-   * @return <code>true</code> if the bundle is <code>STARTED</code>
-   *         <code>false</code> if it is in any other state.
+   * @return An element of {@code STATE_UNINSTALLED},{@code STATE_INSTALLED},
+   *         {@code STATE_RESOLVED}, {@code STATE_STARTING}, {@code STATE_STOPPING},
+   *         {@code STATE_ACTIVE}.
    */
-  bool IsStarted() const;
+  State GetState() const;
 
   /**
    * Returns this bundle's {@link BundleContext}. The returned
    * <code>BundleContext</code> can be used by the caller to act on behalf
    * of this bundle.
    *
-   * <p>
-   * If this bundle is not in the <code>STARTED</code> state, then this
+   * If this bundle is not in the <code>STATE_STARTED</code> state, then this
    * bundle has no valid <code>BundleContext</code>. This method will
    * return <code>nullptr</code> if this bundle has no valid
```

Comment:
This comment is not valid anymore. The returned object is always a BundleContext (valid / invalid).